starfish/experiment/builder/__init__.py

```diff
@@ -35,17 +37,46 @@
 DEFAULT_DIMENSION_ORDER = (Axes.ZPLANE, Axes.ROUND, Axes.CH)
 
 
+class DevNull(io.BytesIO):
+    """A class meant to mimic an open(filepath, 'wb') operation but
+    prevents any actual writing, reading, or seeking.
+
+    This class is an ugly hack to prevent the slicedimage
+    Writer.generate_partition_document() function from needlessly creating
+    a copy of image data.
+
+    See: https://docs.python.org/3/library/io.html
+    See also: cpython/Lib/_pyio.py
+    """
+    def __init__(self, filepath, mode='wb', *args, **kwargs):
+        super(DevNull, self).__init__(*args, **kwargs)
+        self.name = filepath
+        self.mode = mode
+
+    def read(self, size=-1):
+        return bytes(b'')
```

Comment:
I think this should raise an error if it's called.  We're certainly not expecting this to ever be called.

Reply:
Do you have a preference on error type or just: `NotImplementedError`?